src/protobuf-net.BuildTools/Internal/Utils.cs

```diff
@@ -201,5 +216,16 @@ internal static string Qualified(this INamespaceSymbol ns, string type)
             }
             return null;
         }
+
+        internal static CodeGenType GetCodeGenType(this IPropertySymbol symbol)
+        {
+            var symbolTypeName = symbol.Type.Name;
+            if (Enum.TryParse<CodeGenWellKnownType>(symbolTypeName, ignoreCase: true, out var wellKnownType))
+            {
+                return new CodeGenSimpleType.WellKnown(wellKnownType, symbolTypeName, "System.");    
+            }
+
+            return new CodeGenCustomType(symbolTypeName, symbol.Type?.BaseType?.Name ?? string.Empty);
```

Comment:
this is probably overlap with `CodeGenMessage` ? if the difference is that we also need enums: then fine to add a `CodeGenEnum` - although I suspect this might end up virtually the same as `CodeGenMessage`; that might be fine though, and provide useful disambiguation

Reply:
Ye, this is a trouble - I need to resolve types of not parsed symbols.
Will do that tomorrow

Reply:
I agree `Custom` is just bad and does not serve any specific value